common/spl/spl_fit.c

```diff
@@ -580,16 +580,24 @@ static int spl_fit_upload_fpga(struct spl_fit_info *ctx, int node,
 {
 	const char *compatible;
 	int ret;
+	int devnum = 0;
+	int flags = 0;
 
 	debug("FPGA bitstream at: %x, size: %x\n",
 	      (u32)fpga_image->load_addr, fpga_image->size);
 
 	compatible = fdt_getprop(ctx->fit, node, "compatible", NULL);
-	if (!compatible)
+	if (!compatible) {
 		warn_deprecated("'fpga' image without 'compatible' property");
+	} else {
+		flags = fpga_compatible2flag(devnum, compatible);
+		if (strcmp(compatible, "u-boot,fpga-legacy"))
```

Comment:
Since this  "if" is only used by debug should not be enabled only used a debug #ifdef or inside the fpga_compatible2flag?

Reply:
Thanks, good point. Since we add the flag for "u-boot,fpga-legacy", this message seems useless. I'll remove it at all.